src/mlpack/methods/ann/layer/add_merge_impl.hpp

```diff
@@ -0,0 +1,210 @@
+/**
+ * @file methods/ann/layer/add_merge_impl.hpp
+ * @author Shubham Agrawal
+ *
+ * Implementation of the AddMerge class, which acts as a addition container.
+ *
+ * mlpack is free software; you may redistribute it and/or modify it under the
+ * terms of the 3-clause BSD license.  You should have received a copy of the
+ * 3-clause BSD license along with mlpack.  If not, see
+ * http://www.opensource.org/licenses/BSD-3-Clause for more information.
+ */
+#ifndef MLPACK_METHODS_ANN_LAYER_ADD_MERGE_IMPL_HPP
+#define MLPACK_METHODS_ANN_LAYER_ADD_MERGE_IMPL_HPP
+
+#include "add_merge.hpp"
+
+namespace mlpack {
+namespace ann {
+
+template<typename MatType>
+AddMergeType<MatType>::AddMergeType() :
+    MultiLayer<MatType>()
+{
+  // Nothing to do.
+}
+
+template<typename MatType>
+AddMergeType<MatType>::AddMergeType(const AddMergeType& other) :
+    MultiLayer<MatType>(other)
+{
+  // Nothing to do here.
+}
+
+template<typename MatType>
+AddMergeType<MatType>::AddMergeType(AddMergeType&& other) :
+    MultiLayer<MatType>(other)
+{
+  // Nothing to do here.
+}
+
+template<typename MatType>
+AddMergeType<MatType>& AddMergeType<MatType>::operator=(const AddMergeType& other)
+{
+  if (this != &other)
+  {
+    MultiLayer<MatType>::operator=(other);
+  }
+
+  return *this;
+}
+
+template<typename MatType>
+AddMergeType<MatType>& AddMergeType<MatType>::operator=(AddMergeType&& other)
+{
+  if (this != &other)
+  {
+    MultiLayer<MatType>::operator=(other);
+  }
+
+  return *this;
+}
+
+template<typename MatType>
+void AddMergeType<MatType>::Forward(
+    const MatType& input, MatType& output)
+{
+  // Make sure training/testing mode is set right in each layer.
+  for (size_t i = 0; i < this->network.size(); ++i)
+    this->network[i]->Training() = this->training;
+
+  // Note that we use `output` for the last layer; layerOutputs is only used for
+  // intermediate values between layers.
+  if (this->network.size() > 1)
+  {
+    // Initialize memory for the forward pass (if needed).
+    this->InitializeForwardPassMemory(input.n_cols);
+
+    // Forward pass every layer in network with same input.
+    for (size_t i = 0; i < this->network.size(); i++)
+      this->network[i]->Forward(input, this->layerOutputs[i]);
+
+		// Reduce the outputs to single output by adding element-wise.
```

Comment:
```suggestion
        // Reduce the outputs to single output by adding element-wise.
```
Be sure that your editor is using spaces instead of tabs to conform with the style guidelines for mlpack: https://github.com/mlpack/mlpack/wiki/DesignGuidelines#style-guidelines